### PR TITLE
Fix tests to get script/test to pass on Windows

### PR DESCRIPTION
**Pull request by ashmaroli**
- [x] Fix `TestGeneratedSite`
- [x] Fix `TestFilters`
- [x] Fix `TestSite` (only partially)
- [x] Add new helper method to skip testing symlinks on Windows

--

ref: #5241 
/cc @jekyll/windows

### CODE REVIEW

**Comment by ashmaroli** [comment]
This PR is a refined version of the referenced pull.

File: test/test_site.rb [hunk 1]
```diff
@@ -14,14 +14,20 @@ class TestSite < JekyllUnitTest
 
     should "have an array for plugins if passed as a string" do
       site = Site.new(site_configuration({ "plugins_dir" => "/tmp/plugins" }))
-      assert_equal ["/tmp/plugins"], site.plugins
+      array = Utils::Platforms.really_windows? ? ["C:/tmp/plugins"] : ["/tmp/plugins"]
```
Comment (XhmikosR) [inline test/test_site.rb]: Why hardcode this? You should just use the environment variable.

Comment (ashmaroli) [inline test/test_site.rb]: > just use the environment variable

please explain more.

Comment (ashmaroli) [inline test/test_site.rb]: the path is not important here actually. This step and the next is to simply test if **plugins** are correctly detected **as specified**.
e.g. if were to set `plugins_dir => "/apps/bazinga"`, the array returned would be:
```shell
["C:/apps/bazinga"] # on Windows and AppVeyor CI
["/apps/bazinga"] # on Unixoids and Travis CI
```

Comment (XhmikosR) [inline test/test_site.rb]: I'm just saying, it's bad practice hardcoding paths. Especially when you can use something a lot more reliable to detect the path.

Comment (ashmaroli) [inline test/test_site.rb]: the more I think about this, I wonder if this is a bug or intended..
when I set, `plugins_dir => "/apps/plugins"`
I should get a relative path to `plugins` dir inside my site, or an absolute_path to `plugins` dir **with** my site.

Comment (ashmaroli) [inline test/test_site.rb]: moving this segment to a dedicated issue ticket..

File: test/helper.rb [hunk 2]
```diff
@@ -158,4 +158,11 @@ def nokogiri_fragment(str)
       str
     )
   end
+
+  def skip_if_windows(msg = nil)
+    if Utils::Platforms.really_windows?
+      msg ||= "Jekyll doesn't currently support Symlinks on Windows"
```
Comment (parkr) [inline test/helper.rb]: This seems like a strange default. Is this only used in symlinks-related code? If not, I'd suggest something like `Jekyll does not currently support this feature on Windows.`

File: test/test_collections.rb [hunk 3]
```diff
@@ -186,6 +186,9 @@ class TestCollections < JekyllUnitTest
     end
 
     should "include the symlinked file from site.source in the list of docs" do
+      # no support for including symlinked file on Windows
+      skip_if_windows
```
Comment (parkr) [inline test/test_collections.rb]: Then here I'd be specific with my message and say `skip_if_windows, "Jekyll does not currently support symlinsk on Windows."`.

File: test/test_entry_filter.rb [hunk 4]
```diff
@@ -82,6 +82,9 @@ class TestEntryFilter < JekyllUnitTest
 
     # rubocop:disable Performance/FixedSize
     should "include only safe symlinks in safe mode" do
+      # no support for symlinks on Windows
+      skip_if_windows
```
Comment (parkr) [inline test/test_entry_filter.rb]: Same here and everywhere you use this method – make that comment the message instead of a comment so when I run the tests, this information is surfaced.

File: test/test_generated_site.rb [hunk 5]
```diff
@@ -50,11 +50,13 @@ class TestGeneratedSite < JekyllUnitTest
...
+      # adding a pipe character at the beginning preserves formatting with newlines
       expected_output = Regexp.new <<-OUTPUT
-- /css/screen.css last edited at #{time_regexp} with extname .css
-- /pgp.key last edited at #{time_regexp} with extname .key
-- /products.yml last edited at #{time_regexp} with extname .yml
-- /symlink-test/symlinked-dir/screen.css last edited at #{time_regexp} with extname .css
+| - /css/screen.css last edited at #{time_regexp} with extname .css
+  - /pgp.key last edited at #{time_regexp} with extname .key
+  - /products.yml last edited at #{time_regexp} with extname .yml
+  - /symlink-test/symlinked-dir/screen.css last edited at #{time_regexp} with extname .css
```
Comment (parkr) [inline test/test_generated_site.rb]: Why is this necessary? Can you paste the test failure in a comment below?

File: test/test_site.rb [hunk 6]
```diff
@@ -175,7 +175,6 @@ def generate(site)
         method.call(*args, &block).reverse
       end
       @site.process
-      # files in symlinked directories may appear twice
```
Comment (parkr) [inline test/test_site.rb]: Maybe say here than symlinked directories are excluded from this list and added in when not Windows.

File: test/test_generated_site.rb (hunk 5 above)
Comment (ashmaroli) [inline test/test_generated_site.rb]: @parkr :point_down: 
```
Failure:
TestGeneratedSite#test_: generated sites should print a nice list of static files.  [C:/projects/jekyll/test/test_generated_site.rb:59]
Minitest::Assertion: Expected /- \/css\/screen.css last edited at \d+:\d+ with extname .css
- \/pgp.key last edited at \d+:\d+ with extname .key
- \/products.yml last edited at \d+:\d+ with extname .yml
- \/symlink-test\/symlinked-dir\/screen.css last edited at \d+:\d+ with extname .css
/ to match "\n- /css/screen.css last edited at 23:11 with extname .css\n- /pgp.key last edited at 23:11 with extname .key\n- /products.yml last edited at 23:11 with extname .yml\n- /symlink-test/symlinked-dir last edited at 23:11 with extname \n- /symlink-test/symlinked-file last edited at 23:11 with extname \n".
```

**Comment by ashmaroli** [comment]
> Out of curiosity, why are we using really_windows? instead of just windows?

Because `::Platforms.really_windows?` is *more strict* than `::Platforms.windows?`

**Comment by parkr** [comment]
> Because ::Platforms.really_windows? is more strict than ::Platforms.windows?

Why do we want it to be more strict?

**Comment by ashmaroli** [comment]
> Why do we want it to be more strict?

:smiley: Because currently, symlinks is known to not have native support in Windows env. But then it may work in Windows subsystem on Linux

**Comment by ashmaroli** [comment]
TODO: Undo Gemfile changes when `codeclimate-test-reporter` issue is fixed on `master`

**Comment by parkr** [comment]
This is looking excellent!

File: Gemfile [hunk 7]
```diff
@@ -22,7 +22,7 @@ group :test do
   gem "cucumber", "~> 2.1"
   gem "jekyll_test_plugin"
   gem "jekyll_test_plugin_malicious"
-  gem "codeclimate-test-reporter"
+  gem "codeclimate-test-reporter", "0.6.0"
```
Comment (parkr) [inline Gemfile]: Want to submit a separate PR for this?

Comment (ashmaroli) [inline Gemfile]: @parkr, if you think, going with `"~> 0.6.0"` is the better way for now, I shall definitely move this to a dedicated PR. Otherwise, this will remain here to have the tests run correctly..

**Comment by parkr** [comment]
@jekyllbot: merge +dev
